markets/perps-market/contracts/modules/PerpsMarketFactoryModule.sol

```diff
@@ -41,37 +41,23 @@ contract PerpsMarketFactoryModule is IPerpsMarketFactoryModule {
     /**
      * @inheritdoc IPerpsMarketFactoryModule
      */
-    function setSynthetix(ISynthetixSystem synthetix) external override {
-        OwnableStorage.onlyOwner();
-        PerpsMarketFactory.Data storage store = PerpsMarketFactory.load();
-
-        store.synthetix = synthetix;
-        (address usdTokenAddress, ) = synthetix.getAssociatedSystem("USDToken");
-        store.usdToken = ITokenModule(usdTokenAddress);
-        store.oracle = synthetix.getOracleManager();
-    }
-
-    /**
-     * @inheritdoc IPerpsMarketFactoryModule
-     */
-    function setSpotMarket(ISpotMarketSystem spotMarket) external override {
-        OwnableStorage.onlyOwner();
-
-        PerpsMarketFactory.load().spotMarket = spotMarket;
-    }
-
-    /**
-     * @inheritdoc IPerpsMarketFactoryModule
-     */
-    function initializeFactory() external override returns (uint128) {
+    function initializeFactory(
+        ISynthetixSystem synthetix,
+        ISpotMarketSystem spotMarket,
+        string memory marketName
+    ) external override returns (uint128) {
         FeatureFlag.ensureAccessToFeature(_CREATE_MARKET_FEATURE_FLAG);
         OwnableStorage.onlyOwner();
 
         PerpsMarketFactory.Data storage factory = PerpsMarketFactory.load();
 
         uint128 perpsMarketId;
         if (factory.perpsMarketId == 0) {
-            perpsMarketId = factory.synthetix.registerMarket(address(this));
+            setSynthetix(synthetix);
+            setSpotMarket(spotMarket);
+            factory.name = marketName;
+
+            perpsMarketId = synthetix.registerMarket(address(this));
             factory.perpsMarketId = perpsMarketId;
```

Comment:
thoughts on moving all this to `factory.initialize(name, synthetix, spotMarket)` func that returns the `perpsMarketId`?

also in omnibus, this initialize will get re-run and this block will get skipped as expected since the market id exists.  so name will not be able to be set.  maybe add a `setPerpsMarketName` function?